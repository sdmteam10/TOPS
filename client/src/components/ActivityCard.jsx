import React, { useState } from 'react'
import Draggable from 'react-draggable';
import { Card } from 'antd';

  // state = {
  //     activeDrags: 0,
  //     deltaPosition: {
  //       x: 0, y: 0
  //     },
  //     controlledPosition: {
  //       x: -400, y: 200
  //     }
  // }

  

const ActivityCard = ({loadingValue, actName, pos}) => {
const[deltaPosition, setDeltaPosition] = useState(0)
// const handleDrag = (e, ui) => {
//     const {x, y} = this.state.deltaPosition;
//     setDeltaPosition({
//       deltaPosition: {
//         x: x + ui.deltaX,
//         y: y + ui.deltaY,
//       }
//     });
//   };
// onDrag={this.handleDrag}
    return( 
      <Draggable bounds="body">
        <Card hoverable= "true" className="text-center" style={{ width: 200,  marginTop: 16 }} bodyStyle={{  backgroundColor: "#DFFFDF"}} loading={loadingValue}>
         {actName}
        </Card>
        {/* <div>x: {this.state.deltaPosition.x.toFixed(0)}, y: {this.state.deltaPosition.y.toFixed(0)}</div> */}
      </Draggable>
    ) 
}

export default ActivityCard