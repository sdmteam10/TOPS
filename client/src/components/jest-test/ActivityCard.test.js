import React from 'react'
import ActivityCard from '../ActivityCard'
import renderer from 'react-test-renderer';
import { fireEvent, render } from "@testing-library/react"
import axios from "axios"

axios.defaults.adapter = require('axios/lib/adapters/http');


describe('Test <ActivityCard /> component', function() {
  it('Should render', function() {  
    //Render component should be successed.
    const component = renderer.create(
      <ActivityCard />
    );
      let testSnap = component.toJSON();
      expect(testSnap).toMatchSnapshot(); 
  })
  
})
  
      
  
  