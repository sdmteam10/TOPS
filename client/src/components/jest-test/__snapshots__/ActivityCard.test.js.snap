// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ActivityCard /> component Should render 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable text-center react-draggable"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  style={
    Object {
      "marginTop": 16,
      "transform": "translate(0px,0px)",
      "width": 200,
    }
  }
  transform={null}
>
  <div
    className="ant-card-body"
    style={
      Object {
        "backgroundColor": "#DFFFDF",
      }
    }
  />
</div>
`;

exports[`Test <GoalsActivitiesList /> component Should render 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable text-center react-draggable"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  style={
    Object {
      "marginTop": 16,
      "transform": "translate(0px,0px)",
      "width": 200,
    }
  }
  transform={null}
>
  <div
    className="ant-card-body"
    style={
      Object {
        "backgroundColor": "#DFFFDF",
      }
    }
  />
</div>
`;
